test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs

```diff
@@ -39,85 +39,88 @@ public CustomSearchParamTests(HttpIntegrationTestFixture fixture, ITestOutputHel
         [SkippableFact]
         public async Task GivenANewSearchParam_WhenReindexingComplete_ThenResourcesSearchedWithNewParamReturned()
         {
-            var patientName = Guid.NewGuid().ToString().ComputeHash().Substring(28).ToLower();
-            var patient = new Patient { Name = new List<HumanName> { new HumanName { Family = patientName } } };
-            var searchParam = Samples.GetJsonSample<SearchParameter>("SearchParameter");
-            searchParam.Code = "fooCode";
-
-            // POST a new patient
-            FhirResponse<Patient> expectedPatient = await Client.CreateAsync(patient);
-
-            // POST a second patient to show it is filtered and not returned when using the new search parameter
-            await Client.CreateAsync(Samples.GetJsonSample<Patient>("Patient"));
+            var randomName = Guid.NewGuid().ToString().ComputeHash().Substring(0, 14).ToLower();
+            var searchParam = Samples.GetJsonSample<SearchParameter>("SearchParameter-AppointmentStatus");
+            searchParam.Name = randomName;
+            searchParam.Url = searchParam.Url.Replace("foo", randomName);
+            searchParam.Code = randomName + "Code";
+            searchParam.Id = randomName;
+
+            // POST a new appointment
+            var appointment = Samples.GetJsonSample<Appointment>("Appointment");
+            appointment.Status = Appointment.AppointmentStatus.Noshow;
+            var tag = new Coding(null, randomName);
+            appointment.Meta = new Meta();
+            appointment.Meta.Tag.Add(tag);
+            FhirResponse<Appointment> expectedAppointment = await Client.CreateAsync(appointment);
+
+            // POST a second appointment to show it is filtered and not returned when using the new search parameter
+            var appointment2 = Samples.GetJsonSample<Appointment>("Appointment");
```

Comment:
I think it would be nice to explicitly set a different appointment status here (or maybe added a comment which explains that the default Appointment resource has `status=booked`)

Reply:
Explicitly set the value.